core/src/main/scala/org/apache/gearpump/cluster/client/RunningApplication.scala

```diff
@@ -40,6 +44,12 @@ class RunningApplication(val appId: Int, master: ActorRef, timeout: Timeout) {
     }
   }
 
+  def waitUnilFinish(): ApplicationResult = {
```

Comment:
I don't think `ApplicationResult` should be exposed to client since it's an internal interface.